### ISSUE TITLE
Crash (BB #967)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#967]
This issue was migrated from bitbucket.
**Original Reporter:** silvermcd123
**Original Date:** 03.06.2015 22:39:52 GMT+0000
**Original Priority:** major
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/967

<hr>

http://pastebin.com/GL6N2s2r

rev: 17d4f560cb2ad3388df359b626b2baec32861262

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 03.06.2015 22:47:55 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/967/#comment-18576725

<hr>

Hmm, mind changing the line in LootHandler.cpp that looks like

```
#!c++

bool ok_loot = pCreature && (pCreature->IsAlive() == (player->getClass() == CLASS_ROGUE && pCreature->loot.loot_type == LOOT_SKINNING)) || (!pCreature->IsAlive() == (pCreature->GetSkinner() && pCreature->loot.loot_type == LOOT_SKINNING));
```

to

```
#!c++

bool ok_loot = pCreature && pCreature->IsAlive() == (player->getClass() == CLASS_ROGUE && pCreature->loot.loot_type == LOOT_SKINNING);
```

Just be aware you'll break skinning temporarily, but this'll help me determine whether the crash is here or not. Crash log without line numbers is odd!

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** silvermcd123
**Original Date:** 03.06.2015 22:52:11 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/967/#comment-18576772

<hr>

i'm sorry but i'm getting this on a live server, so i'm not sure what it causes or how to reproduce it.
but i will try tomorow, must sleep now.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** silvermcd123
**Original Date:** 03.06.2015 22:54:40 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/967/#comment-18576798

<hr>

A guy just confirmed me, it's a rogue and he's skinning... so , the bug is there! that's how you crash the server.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 03.06.2015 23:46:29 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/967/#comment-18577286

<hr>

What's he skinning exactly? I can't reproduce.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** silvermcd123
**Original Date:** 04.06.2015 08:26:40 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/967/#comment-18583296

<hr>

not sure, but this crashed when he was skinning and since you made that modification for rogues to be able to skin, this problem appeared.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 04.06.2015 09:43:23 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/967/#comment-18585196

<hr>

How'd testing that code go?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** silvermcd123
**Original Date:** 04.06.2015 11:57:30 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/967/#comment-18588012

<hr>

ok i made a copy of the character that was making the crash, with the code you gave me, no crash happened.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** silvermcd123
**Original Date:** 04.06.2015 11:57:54 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/967/#comment-18588022

<hr>

but idd the skinning is broken

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 04.06.2015 12:20:05 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/967/#comment-18588460

<hr>

Fix #967

→ <<cset 7b742a360e0b>>
